### ISSUE TITLE
Different performance when training different transformers version

### ISSUE THREAD

**Issue by quancq** [huggingface/transformers#11772]
## Environment info
- `transformers` version: 4.6 and 4.5
- Platform:
- Python version: 3.7
- PyTorch version (GPU?): 1.8.0 GPU
- Tensorflow version (GPU?):
- Using GPU in script?: yes
- Using distributed or parallel set-up in script?: no

### Who can help

@LysandreJik @sgugger 

Models: PhoBERT (RoBERTa based)
Model hub: https://huggingface.co/vinai/phobert-base/

## Information
Model I am using (PhoBERT ...):

The problem arises when using:
* [x] my own modified scripts: (give details below)

The tasks I am working on is:
* [x] my own task or dataset: (give details below)

## Expected behavior
Training loss, dev loss, dev F1 in each epoch different when training model with transformers version 4.5 and 4.6.

Have anyone meet same this problem?

**Comment by sgugger** [comment]
We do not guarantee the exact reproducibility of training between versions, only with the same version (PyTorch does the same by the way). Are you using the Trainer API? If this is the case, I believe it's the work done to ensure full reproducibility for checkpoints (e.g. you get to the same results training from scratch or resuming from a checkpoint) that is probably creating this difference, as the way the training data was shuffled has been changed.
